File: 0013 罗马数字转整数/main.py
class Solution:
    def romanToInt(self, s: str) -> int:
        Romo ={"M": 1000, "D": 500, "C": 100,
                "L": 50, "X": 10, "V": 5, "I": 1}
        idx=0
        sum=0
        while(idx != len(s)):
            if(idx < len(s)-1 and Romo[s[idx]] <Romo[s[idx+1]] ):
                sum += Romo[s[idx+1]] - Romo[s[idx]]
                idx+=2
            else:
                sum += Romo[s[idx]]
                idx+=1
        return sum

File: 0013 罗马数字转整数/main.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> table={{'M',1000},{'D',500},{'C',100},{'L',50},{'X',10},{'V',5},{'I',1}};
        int lastnum=1000;
        char romanNum=' ';
        int res=0;
        for(int i=0;i<s.size();i++){
            romanNum=s[i];
            if(lastnum < table[ romanNum ])
                res-= lastnum*2;
            res+=table[romanNum];
            lastnum=table[romanNum];
        }
        return res;
    }
};